code/modules/clothing/head/armor.dm

```diff
@@ -305,58 +305,75 @@
 		MATERIAL_GLASS = 10 // glass is reflective yo, make it cost a lot of it - also, visor
 	)
 
-// Riot helmet
-/obj/item/clothing/head/armor/riot
+// toggleable face guard
+/obj/item/clothing/head/armor/faceshield
+	//We cant just use the armor var to store the original since initial(armor) will return a null pointer
+	var/list/armor_up = list(melee = 0, bullet = 0, energy = 0, bomb = 0, bio = 0, rad = 0)
+	var/list/armor_down = list(melee = 0, bullet = 0, energy = 0, bomb = 0, bio = 0, rad = 0)
+
+	var/tint_down = TINT_MODERATE
+	flags_inv = HIDEEARS
+	var/flags_inv_down = HIDEMASK|HIDEEARS|HIDEEYES|HIDEFACE|BLOCKHEADHAIR
+	body_parts_covered = HEAD|EARS
+	var/body_parts_covered_down = HEAD|EARS|EYES|FACE
+	flash_protection = FLASH_PROTECTION_NONE
+	var/flash_protection_down = FLASH_PROTECTION_MAJOR
+	action_button_name = "Flip Face Shield"
+	var/up = FALSE
+	bad_type = /obj/item/clothing/head/armor/faceshield
+
+/obj/item/clothing/head/armor/faceshield/riot
 	name = "riot helmet"
 	desc = "It's a helmet specifically designed to protect against close range attacks."
 	icon_state = "riot"
-	body_parts_covered = HEAD|FACE|EARS
-	var/list/armor_up = list(melee = 35, bullet = 25, energy = 25, bomb = 20, bio = 0, rad = 0)
-	var/list/armor_down = list(melee = 40, bullet = 40, energy = 30, bomb = 35, bio = 0, rad = 0)
+	armor_up = list(melee = 35, bullet = 25, energy = 25, bomb = 20, bio = 0, rad = 0)
+	armor_down = list(melee = 40, bullet = 40, energy = 30, bomb = 35, bio = 0, rad = 0)
 	item_flags = THICKMATERIAL | COVER_PREVENT_MANIPULATION
-	tint = TINT_MODERATE
-	flash_protection = FLASH_PROTECTION_MAJOR
-	action_button_name = "Flip Face Shield"
-	var/up = FALSE
 	price_tag = 150
 	rarity_value = 25
 
-/obj/item/clothing/head/armor/riot/Initialize()
+/obj/item/clothing/head/armor/faceshield/Initialize()
 	. = ..()
-	armor = up ? armor_up : armor_down
-	update_icon()
+	set_is_up(up)
 
-/obj/item/clothing/head/armor/riot/attack_self()
+/obj/item/clothing/head/armor/faceshield/attack_self()
 	toggle()
 
-/obj/item/clothing/head/armor/riot/update_icon()
+/obj/item/clothing/head/armor/faceshield/update_icon()
 	icon_state = up ? "[initial(icon_state)]_up" : initial(icon_state)
 
-/obj/item/clothing/head/armor/riot/verb/toggle()
+//I wanted to name it set_up() but some how I thought that would be misleading
+/obj/item/clothing/head/armor/faceshield/proc/set_is_up(u)
```

Comment:
You can name it as flip_it_up

Reply:
I feel like it should have the word set in it since it's a setter function, also flip_it_up implies that the proc would only flip it up so I don't wana do that, as for flip_at that's okay it's just set_is_up is more clear as to what it's doing

Reply:
Okay, you can rename the u to var/isUp for readability, that's a /proc/, so the decl should reflect that for readability

Reply:
Sure thing!